### ISSUE TITLE
Support for external database

### ISSUE THREAD

**Issue by Ka0o0** [grindsa/acme2certifier#165]
Hi,

if I understood it correctly, acme2certifier currently doesn't have support for external databases, like an externally hosted Postgresql database. It would be nice to have this feature because at least in our case, we already have a database that is being operated by our operations team.

**Comment by Ka0o0** [comment]
Hi,

thank you, indeed this worked. Three things were crucial to get this working:

- Copy everything from the `examples/django` folder into the acme2certifier root folder
- Replace the DB handler with the Django DB handler
- Make sure to have a Django config in `/var/www/acme2certifier/acme2certifier/settings.py`

It was a bit unintuitive tbh. Is there a reason why this is not part of the normal application and just be controllable using a flag in the `acme_srv.cfg`? Then again I don't have any experience with Django or Python.

Another question that arose was if it was possible to scale this service horizontally, e.g. have three instances running in parallel, when they all access the same external DB. Can you think of any implications or problem that might occur? E.g. is there some state within the application that could be problematic? As far as I understood it, the service is stateless, since it's hosted with uWSGI, right? So this would mean it should be easy to scale?

**Comment by grindsa** [comment]
There is always room for improvement especially in the area of documentation :-)

Switching the db_handlers.py is not so easy. I rather prefer to solve this by writing proper instructions.

Horizontal scaling by spinning up new instance accessing the same external database should work without problems as long as you synchronize the config file.  I suggest using the existing docker images as they are stateless by definition.

**Comment by grindsa** [comment]
[improved documentation](https://github.com/grindsa/acme2certifier/blob/master/docs/external_database_support.md) made it into [v0.35](https://github.com/grindsa/acme2certifier/releases/tag/0.35). Thus, closing this issue. In case you have comments feel free to reopen...
